### PR TITLE
Goalie Workflows

### PR DESCRIPTION
**Pull request by benjdlambert**
Added some workflow's to set some labels on reviews. There's another token here which is used for `org:read` only to read the Backstage Organiation to find out members in the review group.

### CODE REVIEW

File: .github/workflows/goalie-on-pr-review-or-response.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            const {data: reviewerMembers} = await github.rest.teams.listMembersInOrg({
+              org: context.payload.organization.login,
+              team_slug: 'reviewers'
+            });
+            return reviewerMembers;
+      - uses: actions/github-script@v5
+        id: fix-labels
+        with:
+          script: |
+            const isReviewer = ${{steps.get-reviewers.outputs.result}}.some(({login}) => login === context.actor);
```
Comment (freben) [inline .github/workflows/goalie-on-pr-review-or-response.yaml:27]: gh actions don't have the problem of treating the `-` in `get-reviewers` as a math minus, do they? :P

File: .github/workflows/goalie-on-pr-review-or-response.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                name: 'awaiting-author'
+              }).catch(() => {});
+              github.rest.issues.addLabels({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                labels: ['awaiting-review']
+              });
```
Comment (freben) [inline .github/workflows/goalie-on-pr-review-or-response.yaml:41]: don't we need to catch the additions too? if i or the author comment twice on the same thing in a row, will this second thing explode?

**Review by freben — APPROVED** [review]

File: .github/workflows/goalie-on-pr-review-or-response.yaml (hunk 1 above)
Comment (benjdlambert) [reply .github/workflows/goalie-on-pr-review-or-response.yaml:27]: Nope works perfectly :joy:

File: .github/workflows/goalie-on-pr-review-or-response.yaml (hunk 2 above)
Comment (benjdlambert) [reply .github/workflows/goalie-on-pr-review-or-response.yaml:41]: Not sure what you mean here?

**Comment by freben** [comment]
Let's look into what this entails for e.g. techdocs / search workflows etc before merging

**Review by freben — APPROVED** [review]

**Review by freben — APPROVED** [review]
awesomesauce!

**Pull request merged**